### DISCUSSION TITLE
Umi & Bigfish 想法征集 06

### DISCUSSION THREAD

**Discussion by sorrycc** [umijs/umi#10821] [Ideas]
> Why this issue? 计划在 Umi Contributor 群增加一条规则。每人每周贡献 2 个改进 Umi 或 Bigfish 的想法（辟起、辟殊、云谦），外包同学减半（1 个），外部 Contributor 随意，我每周三会在群里发个 issue 做收集。

直接回复即可。（含内部敏感信息的可以私发我）

**Comment by sorrycc** [comment]
1、支持 rsc？（想法源自 wakuwork）
2、umi x chatgpt（待会周会分享下）

**Reply by fz6m** [reply]
`umi? xxx` 感觉不错，期待 copilot x 的 docs 能力

**Comment by stormslowly** [comment]
1.  改进 Umi CI 的执行方式  采用构建产物和使用分离的方式，尽量并行执行 
![image](https://user-images.githubusercontent.com/415655/227086664-8bba8943-ddce-4dea-8446-08e47c5496b4.png)

2.  esbuild 压缩对构建产物，做分析，提前发现命名冲突报错，提示用户尝试开启 IIFE  选项

3.  尝试 一键 jest 替换if vitest ，简化测试配置，减少答疑。

**Reply by fz6m** [reply]
免费版没那么多 macos 机器 😭

**Reply by stormslowly** [reply]
可以排队 或者合并 task，这样做了 构建也会让出来很多机器

**Comment by PeachScript** [comment]
1. 实现 `@umijs/swc-plugin-import`，作为 `babel-plugin-import` 的生态平行包（正在联系 `swc-plugin-import` 的作者看能否转让 owner，不行的话就挂 @umijs org 下）
2. checker 支持自动容错，比如：
    a. 遇到没权限的就直接跳过、不计数也不纳入统计结果
    b. 遇到依赖安装失败的自动重试一次

**Reply by sorrycc** [reply]
import 未来可能不会强依赖。

**Comment by hanzebang** [comment]
1. 强约束规则重构（与 doctor 规则结构保持一致），后期可在 doctor 中检查强约束规则状态

**Comment by GSBL** [comment]
有计划支持swc 替换babel吗

**Reply by PeachScript** [reply]
已经支持 swc 了，但短期内不会作为默认值，ref: https://umijs.org/docs/api/config#srctranspiler

**Reply by GSBL** [reply]
umi3.x是没有支持swc的计划了吗

**Reply by PeachScript** [reply]
对，Umi 3 不会增加 feature 了

**Comment by GSBL** [comment]
开启后mfsu设置splictChunks失效有计划处理吗

**Reply by fz6m** [reply]
MFSU 是开发时方案，split chunk 需要仅 prod 运行，自己判断下。
